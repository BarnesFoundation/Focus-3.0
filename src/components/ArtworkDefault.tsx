import React, { Fragment, useEffect, useRef, useState } from "react";
import classnames from "classnames";
import { isAndroid } from "react-device-detect";

import { LANGUAGE_EN } from "../constants";
import withTranslation, {
  LanguageOptionType,
  WithTranslationState,
} from "./withTranslation";
import { ArtworkObject, ArtWorkRecordsResult } from "../types/payloadTypes";
import { EmailForm } from "./EmailForm";
import ProgressiveImage from "react-progressive-image";
import { LanguageDropdown } from "./LanguageDropdown";
import { Share } from "./Share";
import { ContentBlock } from "./ContentBlock";
import google_logo from "../images/google_translate.svg";
import { ScanButton } from "./ScanButton";

type ArtworkDefaultProps = {
  artwork: ArtworkObject;
  result: ArtWorkRecordsResult;
  onSelectLanguage: (selectedLanguage: LanguageOptionType) => void;
  selectedLanguage: LanguageOptionType;
  emailCaptured: boolean;
  showEmailForm: boolean;
  onSubmitEmail: (email: string, callback?: (args?: any) => void) => void;
} & WithTranslationState;

export const ArtworkDefaultComponent: React.FC<ArtworkDefaultProps> = ({
  artwork,
  result,
  onSelectLanguage,
  selectedLanguage,
  emailCaptured,
  showEmailForm,
  onSubmitEmail,
  langOptions,
  getTranslation,
}) => {
  // Component state
  const [imgLoaded, setImgLoaded] = useState(false);
  const [formOpen, setFormOpen] = useState(false);
  const startYCard2 = useRef<number>();
  const startYTrigger = useRef<number>();

  /** Updates state that email was captured and submits it to the server session */
  const handleSubmitEmail = (email: string) => {
    onSubmitEmail(email, () => setFormOpen(false));
  };

  const handleClick = () => {
    if (!formOpen) setFormOpen(true);
  };

  const handleStartSwipe = (event: TouchEvent) => {
    const touch = event.changedTouches[0];
    startYCard2.current = touch.pageY;
  };

  const handleSwipeDismiss = (event: TouchEvent) => {
    const touch = event.changedTouches[0];
    const endY = touch.pageY;
    if (endY > startYCard2.current) setFormOpen(false);
  };

  useEffect(() => {
    if (!imgLoaded || !showEmailForm) return;

    // Add handlers for card-2 swipe down interaction
    const content = document.getElementById("card-content");
    content.addEventListener("touchstart", handleStartSwipe);
    content.addEventListener("touchend", handleSwipeDismiss);
    content.addEventListener("wheel", (e) => {
      if (e.deltaY < 0) {
        setFormOpen(false);
      }
    });

    // Disable scrolling inside element
    const card2 = document.getElementById("card-2");
    card2.addEventListener("touchmove", (e: TouchEvent) => e.preventDefault());

    // Trigger to open card-2 once bottom of content is reached
    const trigger = document.getElementById("card-2__trigger");
    trigger.addEventListener("wheel", (e: WheelEvent) => {
      if (e.deltaY > 0) {
        setFormOpen(true);
      }
    });
    trigger.addEventListener("touchstart", (e: TouchEvent) => {
      startYTrigger.current = e.changedTouches[0].pageY;
    });
    trigger.addEventListener("touchend", (e: TouchEvent) => {
      if (e.changedTouches[0].pageY < startYTrigger.current) setFormOpen(true);
    });
  }, [artwork, imgLoaded, showEmailForm]);

  return (
    <div
      id="container"
      className={classnames(
        "exhibition-obj",
        isAndroid ? "sm-container" : "ios-container"
      )}
    >
      <div id="card-content" className="exhibition-obj__content">
        {!imgLoaded ? (
          <div style={{ visibility: "hidden" }}>
            <img
              className="card-img-result"
              src={artwork.url}
              alt="match_image"
              onLoad={({ target: img }) => setImgLoaded(true)}
            />
          </div>
        ) : (
          <Fragment>
            <div id="card-1" className="exhibition-obj__content__result">
              <div className="row">
                <div className="artwork-top-bg">
                  <img
                    className="card-img-top"
                    src={artwork.bg_url}
                    alt="match_image_background"
                    aria-hidden={true}
                  />
                </div>
                <div className="col-12 col-md-12">
                  <div
                    id="result-card"
                    className="card"
                    data-title={artwork.title}
                    data-artist={artwork.artist}
                    data-id={artwork.id}
                    data-invno={artwork.invno}
                    data-nodesc-invno={
                      !artwork.shortDescription ? artwork.invno : ""
                    }
                  >
                    <div className="card-top-container">
                      <div className="card-img-overlay">
                        <div className="card-header h1">Focused Artwork</div>
                        <div className="card-img-result">
                          {/* @ts-ignore */}
                          <ProgressiveImage
                            src={artwork.url}
                            placeholder={artwork.url_low_quality}
                          >
                            {(src) => (
                              <img
                                src={src}
                                alt="match_image"
                                role="img"
                                aria-label={`${artwork.title} by ${
                                  artwork.artist
                                }${
                                  artwork.culture
                                    ? `, ${artwork.culture}.`
                                    : "."
                                } ${artwork.visualDescription}`}
                              />
                            )}
                          </ProgressiveImage>
                        </div>
                        <div className="card-artist">{artwork.artist}</div>
                        <div className="card-title">{artwork.title}</div>
                      </div>
                    </div>
                    <div className="card-body" id="focussed-artwork-body">
                      <div className="share-wrapper">
                        {/* Language options button */}
                        <div className="language-dropdown-wrapper">
                          <div className="language-dropdown">
                            <LanguageDropdown
                              langOptions={langOptions}
                              selected={selectedLanguage}
                              onSelectLanguage={onSelectLanguage}
                            />
                          </div>
                        </div>

                        {/* Share options button for Barnes Collection objects */}
                        {/* TODO: determine if we want share button for special exhibition objects */}
                        {!result.data.specialExhibition && (
                          <Share
                            shareText={getTranslation("Result_page", "text_1")}
                            artwork={artwork}
                          />
                        )}
                      </div>

                      <div className="short-desc-container">
                        {/* Content blocks from Hygraph CMS */}
                        {artwork.content && (
                          <div className="card-content">
                            {artwork.content.map((c, index) => (
                              <ContentBlock
                                contentBlock={c.contentBlock}
                                key={index}
                              />
                            ))}
                          </div>
                        )}
                      </div>

                      {artwork.content &&
                        selectedLanguage.code !== LANGUAGE_EN && (
                          <div className="google-translate-disclaimer">
                            <span>Translated with </span>
                            <img src={google_logo} alt="google_logo" />
                          </div>
                        )}

                      <div className="card-info">
                        <table className="detail-table">
                          <tbody>
                            <tr>
                              <td className="text-left item-label">{`${getTranslation(
                                "Result_page",
                                "text_3"
                              )}:`}</td>
                              <td className="text-left item-info">
                                {artwork.artist}{" "}
                                {!artwork.unIdentified && artwork.nationality
                                  ? `(${artwork.nationality}, ${artwork.birthDate} - ${artwork.deathDate})`
                                  : ""}
                              </td>
                            </tr>
                            {artwork.unIdentified && (
                              <tr>
                                <td className="text-left item-label">
                                  {`${getTranslation(
                                    "Result_page",
                                    "text_10"
                                  )}:`}
                                </td>
                                <td className="text-left item-info">
                                  {artwork.culture}
                                </td>
                              </tr>
                            )}
                            <tr>
                              <td className="text-left item-label">{`${getTranslation(
                                "Result_page",
                                "text_4"
                              )}:`}</td>
                              <td className="text-left item-info">
                                {artwork.title}
                              </td>
                            </tr>
                            <tr>
                              <td className="text-left item-label">{`${getTranslation(
                                "Result_page",
                                "text_5"
                              )}:`}</td>
                              <td className="text-left item-info">
                                {artwork.displayDate}
                              </td>
                            </tr>
                            <tr>
                              <td className="text-left item-label">{`${getTranslation(
                                "Result_page",
                                "text_6"
                              )}:`}</td>
                              <td className="text-left item-info">
                                {artwork.medium}
                              </td>
                            </tr>
                            <tr>
                              <td className="text-left item-label">{`${getTranslation(
                                "Result_page",
                                "text_7"
                              )}:`}</td>
                              <td className="text-left item-info">
                                {artwork.dimensions}
                              </td>
                            </tr>
                            {!artwork.curatorialApproval && (
                              <tr>
                                <td className="text-left item-label">
                                  {`${getTranslation(
                                    "Result_page",
                                    "text_8"
                                  )}:`}
                                </td>
                                <td className="text-left item-info">
                                  {getTranslation("Result_page", "text_9")}
                                </td>
                              </tr>
                            )}
                          </tbody>
                        </table>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div id="card-2__trigger" />
            </div>
            {showEmailForm ? (
              <div
                id="card-2"
                className={classnames(
                  "exhibition-obj__content__email",
                  "card",
                  {
                    open: formOpen,
                    completed: !formOpen && emailCaptured,
                  }
                )}
                onClick={handleClick}
              >
                <EmailForm
                  withStory={false}
                  isEmailScreen={false}
                  onSubmitEmail={handleSubmitEmail}
                  getTranslation={getTranslation}
                  getSize={(height) => null}
                  pointerEvents="auto"
                  handleClickScroll={(storyIndex, isStoryCard) => null}
                  alwaysFloatBtn={true}
                />
              </div>
            ) : (
              <ScanButton float={false} />
            )}
          </Fragment>
        )}
      </div>
    </div>
  );
};

export const ArtworkDefault = withTranslation<ArtworkDefaultProps>(
  ArtworkDefaultComponent
);
