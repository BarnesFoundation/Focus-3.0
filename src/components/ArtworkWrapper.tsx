import React, { Fragment, useEffect, useRef, useState } from "react";
import { useLocation } from "react-router-dom";
import classnames from "classnames";
import { isTablet } from "react-device-detect";
import { isAndroid } from "react-device-detect";

import { SearchRequestService } from "../services/SearchRequestService";
import * as constants from "../constants";
import withTranslation, {
  LanguageOptionType,
  WithTranslationState,
} from "./withTranslation";
import {
  constructResultAndInRoomSlider,
  constructStory,
} from "../helpers/artWorkHelper";
import {
  ArtworkObject,
  ArtWorkRecordsResult,
  StoryResponse,
} from "../types/payloadTypes";
import { useLocalStorage } from "../hooks/useLocalStorage";
import { EmailForm } from "./EmailForm";
import ProgressiveImage from "react-progressive-image";
import { LanguageDropdown } from "./LanguageDropdown";
import { Share } from "./Share";
import { ContentBlock } from "./ContentBlock";
import google_logo from "../images/google_translate.svg";
import { ScanButton } from "./ScanButton";
import Artwork from "./Artwork";
import { ExhibitionObject } from "./ExhibitionObject";

export const ArtworkWrapperComponent: React.FC<WithTranslationState> = ({
  getSelectedLanguage,
}) => {
  // Initialize the search request services
  const sr = new SearchRequestService();
  // Component state
  const [imageId, setImageId] = useState<string>();
  const [artwork, setArtwork] = useState<ArtworkObject["artwork"]>();
  const [result, setResult] = useState<ArtWorkRecordsResult>();
  const [stories, setStories] = useState<StoryResponse>();
  const [imgLoaded, setImgLoaded] = useState(false);
  const [emailCaptured, setEmailCaptured] = useState(false);
  const [showEmailForm, setShowEmailForm] = useState(true);
  const [emailCaptureAck, setEmailCaptureAck] = useState(false);
  const [selectedLanguage, setSelectedLanguage] = useState<LanguageOptionType>({
    name: "English",
    code: "En",
    selected: true,
  });

  const location = useLocation();
  const { getLocalStorage } = useLocalStorage();

  const [formOpen, setFormOpen] = useState(false);
  const startYCard2 = useRef<number>();
  const startYTrigger = useRef<number>();

  const setupStory = async (imageId) => {
    const storyInformation = await sr.getStoryItems(imageId);
    return constructStory(storyInformation);
  };

  // Set state and get object data on component initialization
  useEffect(() => {
    const componentWillMount = async () => {
      let artworkInfo: ArtWorkRecordsResult;
      let storyResponse: StoryResponse;
      let imageId: string;

      // Check if result data is stored in location state from camera component
      if (location.state?.result) {
        imageId = location.state.result.data.records[0].id;
        artworkInfo = location.state.result;
      } else {
        // If result data is not in state, get image ID from path and get data from server
        const [_n, path, id] = location.pathname.split("/");
        imageId = id;

        if (path === "artwork") {
          [artworkInfo, storyResponse] = await Promise.all([
            sr.getArtworkInformation(imageId),
            setupStory(imageId),
          ]);
        } else if (path === "exhibition") {
          artworkInfo = await sr.getSpecialExhibitionObject(imageId);
        }
      }

      const { artwork, roomRecords } = constructResultAndInRoomSlider(
        artworkInfo,
        isTablet
      );

      setResult(artworkInfo);
      setArtwork(artwork);
      setStories(storyResponse);
      setImageId(imageId);

      const emailRecorded = getLocalStorage(constants.SNAP_USER_EMAIL) !== null;
      setEmailCaptured(emailRecorded);
      setShowEmailForm(!emailRecorded);
      setEmailCaptureAck(emailRecorded);
      setSelectedLanguage((await getSelectedLanguage())[0]);
    };

    componentWillMount();
  }, []);

  return (
    <Fragment>
      {result &&
        artwork &&
        (result.data.showStory ? (
          <Artwork />
        ) : (
          <ExhibitionObject
            initArtwork={artwork}
            initResult={result}
            imageId={imageId}
          />
        ))}
    </Fragment>
  );
};

export const ArtworkWrapper = withTranslation(ArtworkWrapperComponent);
