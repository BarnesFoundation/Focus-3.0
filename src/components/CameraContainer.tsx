import React, { Component } from "react";
import { withRouter, History } from "react-router-dom";
import { compose } from "redux";

import posed from "react-pose";

import Camera from "./Camera";
import { StorableSearch, ImageSearchResponse } from "../classes/searchResponse";
import { SearchRequestService } from "../services/SearchRequestService";
import { loadImage } from "./CameraHelper";
import * as constants from "../constants";
import { ARTWORK, EXHIBITION } from "../constants/routes";
import withTranslation, { WithTranslationState } from "./withTranslation";

const Container = posed.div({
  enter: { opacity: 1 },
  exit: { opacity: 0 },
});

type CameraContainerProps = {
  history: History;
} & WithTranslationState;

type CameraContainerState = {
  scanSeqId: any;
  shouldBeScanning: boolean;
  sessionYieldedMatch: boolean;
};

export class CameraContainerComponent extends Component<
  CameraContainerProps,
  CameraContainerState
> {
  sr: SearchRequestService;
  snapAttempts: number;
  responseCounter: number;

  constructor(props) {
    super(props);

    this.sr = new SearchRequestService();
    this.snapAttempts =
      parseInt(localStorage.getItem(constants.SNAP_ATTEMPTS)) || 0;

    this.state = {
      scanSeqId: null,
      shouldBeScanning: null,
      sessionYieldedMatch: null,
    };
  }

  /** Starts a scanning session over a 3-second duration */
  beginScanning = () => {
    // Update the snap attempts with this scan as a single attempt
    localStorage.setItem(
      constants.SNAP_ATTEMPTS,
      (this.snapAttempts + 1).toString()
    );
    localStorage.setItem(constants.SNAP_LAST_TIMESTAMP, Date.now().toString());

    // Reset scans taken and stop any existing scanning
    this.responseCounter = 0;
    this.snapAttempts = this.snapAttempts + 1;
    this.setState({
      shouldBeScanning: true,
      scanSeqId: Date.now(),
      sessionYieldedMatch: null,
    });
  };

  /** Executes steps for completing a scan from an image capture to determining a match */
  processImageCapture = async (imageBlob) => {
    // Only process image capture when we haven't determined if session yielded a match
    if (this.state.sessionYieldedMatch === null && this.responseCounter <= 9) {
      // Prepare and send the request to Catchoom for a response
      const imageSearchRequestConfig = this.sr.prepareRequest(
        imageBlob,
        this.state.scanSeqId
      );
      const imageSearchResponse = await this.sr.submitImageSearchRequest(
        imageSearchRequestConfig
      );

      const { data } = imageSearchRequestConfig;
      const { searchWasSuccessful, searchTime } = imageSearchResponse;
      let searchResultToStore,
        elasticSearchResponse = null;

      // Increment the response counter
      this.responseCounter++;

      // If search was successful
      if (searchWasSuccessful && this.state.shouldBeScanning) {
        this.setState({ shouldBeScanning: false, sessionYieldedMatch: true });

        // Get the identified image information
        const identifiedItem = imageSearchResponse.responsePayload.results[0];

        const identifiedImageInformation = await this.sr.processIdentifiedItem(
          identifiedItem
        );

        const { esResponse, referenceImageUrl } = identifiedImageInformation;

        searchResultToStore = new StorableSearch(
          data,
          searchWasSuccessful,
          referenceImageUrl,
          esResponse,
          searchTime
        );
        elasticSearchResponse = esResponse;

        if (elasticSearchResponse) {
          this.completeImageSearchRequest(elasticSearchResponse);
        } else {
           // The ES response for special exhibition objects can be undefined if no content has bee added to Hygraph for the object,
          //  In that case we will want to end scanning by updating the state to show that the search did not yield a match
          this.setState({
            sessionYieldedMatch: false,
          });
        }
      }

      // Otherwise, when it's not successful
      else {
        searchResultToStore = new StorableSearch(
          data,
          searchWasSuccessful,
          null,
          elasticSearchResponse,
          searchTime
        );
      }

      // Store the result, regardless of success or not
      this.sr.storeSearchedResult(searchResultToStore);

      // If we've received 9 responses and haven't completed the search, the session yielded no match
      if (
        this.responseCounter === 9 &&
        this.state.sessionYieldedMatch == null &&
        this.state.shouldBeScanning
      ) {
        // Updates the state to show that the search did not yield a match and scanning should end
        this.setState({ shouldBeScanning: false, sessionYieldedMatch: false });
      }
    }
  };

  /** Processes the completion of successful image search */
  completeImageSearchRequest(response) {
    const { width } = screen;

    // Get the record and art url from it
    const record = response["data"]["records"][0];
    const { art_url: artUrl, id } = record;
    const specialExhibition: boolean = response["data"]["specialExhibition"];

    // Load the image background first so that it gets cached for faster displaying
    const matchImage = loadImage(`${artUrl}?w=${width - 80}`);
    const matchImageBg = loadImage(
      `${artUrl}?q=0&auto=compress&crop=faces,entropy&fit=crop&w=${width}`
    );

    Promise.all([matchImage, matchImageBg]).then(() => {
      // Navigate to the artwork or exhibition page
      this.props.history.push({
        pathname: `${specialExhibition ? EXHIBITION : ARTWORK}/${id}`,
        state: { result: response },
      });
    });
  }

  componentDidMount() {
    this.beginScanning();
  }

  render() {
    const { processImageCapture, snapAttempts, beginScanning } = this;
    const { sessionYieldedMatch, shouldBeScanning } = this.state;

    return (
      <Container className="camera-container" initialPose="exit" pose="enter">
        <Camera
          // @ts-ignore
          processImageCapture={processImageCapture}
          sessionYieldedMatch={sessionYieldedMatch}
          beginScanning={beginScanning}
          snapAttempts={snapAttempts}
          shouldBeScanning={shouldBeScanning}
        />
      </Container>
    );
  }
}

export default compose(withRouter, withTranslation)(CameraContainerComponent);
