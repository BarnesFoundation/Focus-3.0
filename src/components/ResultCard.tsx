import React from "react";
import ProgressiveImage from "react-progressive-image";
import { LANGUAGE_EN, SNAP_LANGUAGE_PREFERENCE } from "../constants";
import { useLocalStorage } from "../hooks/useLocalStorage";
import google_logo from "../images/google_translate.svg";
import { WithTranslationState } from "../types";
import { LanguageDropdown } from "./LanguageDropdown";
import { Share } from "./Share";
import withTranslation from "./withTranslation";

// TODO: add in type defs as we refactor parent components
type ResultCardProps = {
  artwork;
  refCallbackInfo;
  setArtworkRef;
  langOptions;
  selectedLanguage;
  onSelectLanguage;
  shortDescContainer;
  specialExhibition: boolean;
} & WithTranslationState;

export const ResultCardComponent: React.FC<ResultCardProps> = ({
  artwork,
  refCallbackInfo,
  setArtworkRef,
  langOptions,
  selectedLanguage,
  onSelectLanguage,
  shortDescContainer,
  specialExhibition,
  getTranslation,
}) => {
  const { getLocalStorage } = useLocalStorage();

  const shortDescFontStyle =
    getLocalStorage(SNAP_LANGUAGE_PREFERENCE) === "Ru"
      ? { fontSize: `14px` }
      : {};

  return (
    <div className="container-fluid artwork-container" id="search-result">
      <div className="row" ref={refCallbackInfo}>
        <div className="artwork-top-bg">
          <img
            className="card-img-top"
            src={artwork.bg_url}
            alt="match_image_background"
            aria-hidden={true}
          />
        </div>
        <div className="col-12 col-md-12">
          <div
            id="result-card"
            className="card"
            data-title={artwork.title}
            data-artist={artwork.artist}
            data-id={artwork.id}
            data-invno={artwork.invno}
            data-nodesc-invno={!artwork.shortDescription ? artwork.invno : ""}
          >
            <div className="card-top-container">
              <div className="card-img-overlay">
                <div className="card-header h1">Focused Artwork</div>
                <div className="card-img-result">
                  {/* @ts-ignore */}
                  <ProgressiveImage
                    src={artwork.url}
                    placeholder={artwork.url_low_quality}
                  >
                    {(src) => (
                      <img
                        src={src}
                        alt="match_image"
                        role="img"
                        aria-label={`${artwork.title} by ${artwork.artist}${
                          artwork.culture ? `, ${artwork.culture}.` : "."
                        } ${artwork.visualDescription}`}
                      />
                    )}
                  </ProgressiveImage>
                  {/* <img src={artwork.url} alt="match_image" /> */}
                </div>
                <div className="card-artist">{artwork.artist}</div>
                <div className="card-title">{artwork.title}</div>
              </div>
            </div>
            <div
              className="card-body"
              id="focussed-artwork-body"
              ref={setArtworkRef}
            >
              <div className="share-wrapper">
                {/* Language options button */}
                <div className="language-dropdown-wrapper">
                  <div className="language-dropdown">
                    <LanguageDropdown
                      langOptions={langOptions}
                      selected={selectedLanguage}
                      onSelectLanguage={onSelectLanguage}
                    />
                  </div>
                </div>

                {/* Share options button for Barnes Collection objects */}
                {/* TODO: determine if we want share button for special exhibition objects */}
                {!specialExhibition && (
                  <Share
                    shareText={getTranslation("Result_page", "text_1")}
                    artwork={artwork}
                  />
                )}
              </div>

              <div className="short-desc-container" ref={shortDescContainer}>
                {artwork.shortDescription && (
                  <div
                    className="card-text paragraph"
                    style={shortDescFontStyle}
                    dangerouslySetInnerHTML={{
                      __html: artwork.shortDescription,
                    }}
                  ></div>
                )}
              </div>
              {artwork.shortDescription &&
                selectedLanguage.code !== LANGUAGE_EN && (
                  <div className="google-translate-disclaimer">
                    <span>Translated with </span>
                    <img src={google_logo} alt="google_logo" />
                  </div>
                )}
              <div className="card-info">
                <table className="detail-table">
                  <tbody>
                    <tr>
                      <td className="text-left item-label">{`${getTranslation(
                        "Result_page",
                        "text_3"
                      )}:`}</td>
                      <td className="text-left item-info">
                        {artwork.artist}{" "}
                        {!artwork.unIdentified && artwork.nationality
                          ? `(${artwork.nationality}, ${artwork.birthDate} - ${artwork.deathDate})`
                          : ""}
                      </td>
                    </tr>
                    {artwork.unIdentified && (
                      <tr>
                        <td className="text-left item-label">
                          {`${getTranslation("Result_page", "text_10")}:`}
                        </td>
                        <td className="text-left item-info">
                          {artwork.culture}
                        </td>
                      </tr>
                    )}
                    <tr>
                      <td className="text-left item-label">{`${getTranslation(
                        "Result_page",
                        "text_4"
                      )}:`}</td>
                      <td className="text-left item-info">{artwork.title}</td>
                    </tr>
                    <tr>
                      <td className="text-left item-label">{`${getTranslation(
                        "Result_page",
                        "text_5"
                      )}:`}</td>
                      <td className="text-left item-info">
                        {artwork.displayDate}
                      </td>
                    </tr>
                    <tr>
                      <td className="text-left item-label">{`${getTranslation(
                        "Result_page",
                        "text_6"
                      )}:`}</td>
                      <td className="text-left item-info">{artwork.medium}</td>
                    </tr>
                    <tr>
                      <td className="text-left item-label">{`${getTranslation(
                        "Result_page",
                        "text_7"
                      )}:`}</td>
                      <td className="text-left item-info">
                        {artwork.dimensions}
                      </td>
                    </tr>
                    {!artwork.curatorialApproval && (
                      <tr>
                        <td className="text-left item-label">
                          {`${getTranslation("Result_page", "text_8")}:`}
                        </td>
                        <td className="text-left item-info">
                          {getTranslation("Result_page", "text_9")}
                        </td>
                      </tr>
                    )}
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export const ResultCard = withTranslation(ResultCardComponent);
