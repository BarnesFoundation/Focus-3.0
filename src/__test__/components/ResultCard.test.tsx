import React from "react";
import { render, cleanup } from "@testing-library/react";
import { ResultCard } from "../../components/ResultCard";

afterEach(cleanup);

describe("ResultCard", () => {
  it("should match the snapshot for a Barnes collection object", () => {
    const { container } = render(
      <ResultCard
        getTranslation={(screen, textId) => textId + " " + screen}
        langOptions={[]}
        artwork={{
          bg_url: "bg_url",
          title: "title",
          artist: "artist",
          id: 123,
          invno: "invno",
        }}
        refCallbackInfo={(element: any) => null}
        setArtworkRef={(element: any) => null}
        selectedLanguage={{ code: null, name: null, selected: true }}
        onSelectLanguage={() => null}
        specialExhibition={false}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot for a special exhibition object", () => {
    const { container } = render(
      <ResultCard
        getTranslation={(screen, textId) => textId + " " + screen}
        langOptions={[]}
        artwork={{
          bg_url: "bg_url",
          title: "title",
          artist: "artist",
          id: 123,
          invno: "invno",
        }}
        refCallbackInfo={(element: any) => null}
        setArtworkRef={(element: any) => null}
        selectedLanguage={{ code: null, name: null, selected: true }}
        onSelectLanguage={() => null}
        specialExhibition={false}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
