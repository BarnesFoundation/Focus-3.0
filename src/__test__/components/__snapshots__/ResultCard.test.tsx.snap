// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCard should match the snapshot for a Barnes collection object 1`] = `
<div>
  <div
    class="container-fluid artwork-container"
    id="search-result"
  >
    <div
      class="row"
    >
      <div
        class="artwork-top-bg"
      >
        <img
          alt="match_image_background"
          aria-hidden="true"
          class="card-img-top"
          src="bg_url"
        />
      </div>
      <div
        class="col-12 col-md-12"
      >
        <div
          class="card"
          data-artist="artist"
          data-id="123"
          data-invno="invno"
          data-nodesc-invno="invno"
          data-title="title"
          id="result-card"
        >
          <div
            class="card-top-container"
          >
            <div
              class="card-img-overlay"
            >
              <div
                class="card-header h1"
              >
                Focused Artwork
              </div>
              <div
                class="card-img-result"
              >
                <img
                  alt="match_image"
                  aria-label="title by artist. undefined"
                  role="img"
                />
              </div>
              <div
                class="card-artist"
              >
                artist
              </div>
              <div
                class="card-title"
              >
                title
              </div>
            </div>
          </div>
          <div
            class="card-body"
            id="focussed-artwork-body"
          >
            <div
              class="share-wrapper"
            >
              <div
                class="language-dropdown-wrapper"
              >
                <div
                  class="language-dropdown"
                >
                  <div
                    class="dd-wrapper"
                  >
                    <div
                      aria-haspopup="true"
                      class="dd-header"
                      id="language-btn"
                    >
                      <div
                        aria-labelledby="language-btn"
                        class="dd-header-title"
                      />
                      <span>
                        <img
                          aria-hidden="true"
                          aria-label="Down"
                          src="down_gray_1x.png"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="btn-share-result"
                  id="share-it"
                >
                  <img
                    alt="share"
                    src="share-icon.svg"
                  />
                  <span
                    class="text-share"
                  >
                    text_1 Result_page
                  </span>
                </div>
              </div>
            </div>
            <div
              class="short-desc-container"
            />
            <div
              class="card-info"
            >
              <table
                class="detail-table"
              >
                <tbody>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_3 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    >
                      artist
                       
                      
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_4 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    >
                      title
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_5 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    />
                  </tr>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_6 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    />
                  </tr>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_7 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    />
                  </tr>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_8 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    >
                      text_9 Result_page
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ResultCard should match the snapshot for a special exhibition object 1`] = `
<div>
  <div
    class="container-fluid artwork-container"
    id="search-result"
  >
    <div
      class="row"
    >
      <div
        class="artwork-top-bg"
      >
        <img
          alt="match_image_background"
          aria-hidden="true"
          class="card-img-top"
          src="bg_url"
        />
      </div>
      <div
        class="col-12 col-md-12"
      >
        <div
          class="card"
          data-artist="artist"
          data-id="123"
          data-invno="invno"
          data-nodesc-invno="invno"
          data-title="title"
          id="result-card"
        >
          <div
            class="card-top-container"
          >
            <div
              class="card-img-overlay"
            >
              <div
                class="card-header h1"
              >
                Focused Artwork
              </div>
              <div
                class="card-img-result"
              >
                <img
                  alt="match_image"
                  aria-label="title by artist. undefined"
                  role="img"
                />
              </div>
              <div
                class="card-artist"
              >
                artist
              </div>
              <div
                class="card-title"
              >
                title
              </div>
            </div>
          </div>
          <div
            class="card-body"
            id="focussed-artwork-body"
          >
            <div
              class="share-wrapper"
            >
              <div
                class="language-dropdown-wrapper"
              >
                <div
                  class="language-dropdown"
                >
                  <div
                    class="dd-wrapper"
                  >
                    <div
                      aria-haspopup="true"
                      class="dd-header"
                      id="language-btn"
                    >
                      <div
                        aria-labelledby="language-btn"
                        class="dd-header-title"
                      />
                      <span>
                        <img
                          aria-hidden="true"
                          aria-label="Down"
                          src="down_gray_1x.png"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="btn-share-result"
                  id="share-it"
                >
                  <img
                    alt="share"
                    src="share-icon.svg"
                  />
                  <span
                    class="text-share"
                  >
                    text_1 Result_page
                  </span>
                </div>
              </div>
            </div>
            <div
              class="short-desc-container"
            />
            <div
              class="card-info"
            >
              <table
                class="detail-table"
              >
                <tbody>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_3 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    >
                      artist
                       
                      
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_4 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    >
                      title
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_5 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    />
                  </tr>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_6 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    />
                  </tr>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_7 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    />
                  </tr>
                  <tr>
                    <td
                      class="text-left item-label"
                    >
                      text_8 Result_page:
                    </td>
                    <td
                      class="text-left item-info"
                    >
                      text_9 Result_page
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
